sdk/identity/Azure.Identity/src/MsalPublicClient.cs

```diff
@@ -39,49 +38,53 @@ protected override Task<IPublicClientApplication> CreateClientAsync()
                 pubAppBuilder = pubAppBuilder.WithRedirectUri(_redirectUrl);
             }
 
-            return Task.FromResult(pubAppBuilder.Build());
+            return new ValueTask<IPublicClientApplication>(pubAppBuilder.Build());
         }
 
-        public virtual async Task<IEnumerable<IAccount>> GetAccountsAsync()
+        public virtual async ValueTask<List<IAccount>> GetAccountsAsync(bool async, CancellationToken cancellationToken)
         {
-            await EnsureInitializedAsync(true).ConfigureAwait(false);
-
-            return await Client.GetAccountsAsync().ConfigureAwait(false);
+            IPublicClientApplication client = await GetClientAsync(async, cancellationToken).ConfigureAwait(false);
+            return await AcquireTokenSilent(client, async).ConfigureAwait(false);
```

Comment:
Shouldn't this be calling `GetAccounts` instead of `AcquireTokenSilent`?

Reply:
It calls this method: https://github.com/Azure/azure-sdk-for-net/pull/14156/files/75593183e4d3dc7b546601a797f32a9d0c001fdf#diff-1969a93d707069514862dbbcd1cef73aR80
I will rename it.

Reply:
Fixed